src/codecs/webp/extended.rs

```diff
@@ -536,6 +536,23 @@ struct AnimatedFrame {
     image: WebPStatic,
 }
 
+/// Reads a chunk, but silently ignores unknown chunks at the end of a file
+fn read_extended_chunk<R>(r: &mut R) -> ImageResult<Option<(Cursor<Vec<u8>>, WebPRiffChunk)>>
+where
+    R: Read,
+{
+    let mut unknown_chunk = Ok(());
+
+    loop {
+        match read_chunk(r) {
+            Ok(None) => return Ok(None),
+            Ok(chunk) => return unknown_chunk.and(Ok(chunk)),
+            Err(err @ ImageError::Decoding(_)) => unknown_chunk = unknown_chunk.and(Err(err)),
```

Comment:
Isn't this going to catch all chunks that failed to decode, regardless of whether they're of known or unknown chunk types?

Reply:
Having an unknown chunk type is, at least currently, the only way a chunk can fail to decode with `ImageError::Decoding`, since the chunk's contents aren't decoded at this point.

Reply:
Probably best if it's split in a separate function (like `read_chunk_inner`) to prevent changing `read_chunk`'s signature that's also used elsewhere, but it should be doable. I'll have a look later.